diff --git a/kernel/kmod.c b/kernel/kmod.c
index ae5736e..e0997cd 100644
--- a/kernel/kmod.c
+++ b/kernel/kmod.c
@@ -276,8 +276,12 @@ static int ____call_usermodehelper(void *data)
 	   out the path to be used prior to this point and are now operating
 	   on that copy
 	*/
-	if (strncmp(sub_info->path, "/sbin/", 6) || strstr(sub_info->path, "..")) {
-		printk(KERN_ALERT "grsec: denied exec of usermode helper binary %.950s located outside of /sbin\n", sub_info->path);
+	int   p = strncmp(sub_info->path, "/sbin/", 6);
+	int   q = strncmp(sub_info->path, "/usr/lib/systemd/", 17);
+	char* r =  strstr(sub_info->path, "..");
+
+	if ((p || q) && !(p && q) || r) {
+		printk(KERN_ALERT "grsec: denied exec of usermode helper binary %.950s located outside of /sbin or /usr/lib/systemd\n", sub_info->path);
 		retval = -EPERM;
 		goto fail;
 	}
